File: problems/cpp/1_Two-sum/1.two-sum.cpp
// @before-stub-for-debug-begin
#include <vector>
#include <string>
#include "commoncppproblem1.h"

using namespace std;
// @before-stub-for-debug-end

/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
#include <unordered_map>
#include <vector>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {
        std::unordered_map<int, int> numToIndex;

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            int complement = target - num;
            
            if (numToIndex.find(complement) != numToIndex.end()) 
                return {numToIndex[complement], i};
            numToIndex[num] = i;
        }

        return {};
    }
};
// @lc code=end

File: problems/python/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#
# @lc code=start
from ast import List

class Solution: 
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, num in enumerate(nums):
            com = target - num
            if com in map:
                return [map[com], i]
            
            map[num] = i
            
        return []
        
# @lc code=end

